### PR TITLE
Refactor gphdfs code that depends on Java 1.7 construct.

### PR DESCRIPTION
**Pull request by asimrp**
We have conflicting dependencies in gpdb code today.  gphdfs requires
JDK 1.7 to compile.  However, pljava fails to compile with JDK 1.7 and
requires JDK 1.6.  In order to be consistent with regards to JDK
requirement, this commit removes the construct in gpdhfs that was
newly introduced in JDK 1.7.  At some point, we should unshackle
pljava from JDK 1.6.

### CODE REVIEW

**Comment by asimrp** [comment]
@yydzero @randomtask1155 can you guys please take a look?

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java [hunk 1]
```diff
@@ -139,18 +140,21 @@ protected static void loginSecureHadoop(Configuration conf) throws IOException,
...
+		 * Fail back to securityutil if kinit is disabled.  If
+		 * kinit is not disabled then do nothing because we
+		 * will request a new TGT and update the ticket cache
+		 * regardless if login or kinit refresh failed
+		 * initially.
+		 */
+		if ( kinitDisabled &&
+		     (ex == null || ex instanceof LoginException ||
+		      ex instanceof InterruptedException) ) { // login from keytab
 			SecurityUtil.login(conf,HADOOP_SECURITY_USER_KEYTAB_FILE,HADOOP_SECURITY_USERNAME);
```
Comment (yydzero) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: This is different with original logic, in original code, if kinitDisabled, then no matter there is exception or not, or no matter the exception is LoginException, or InterruptedException or other exception, SecurityUtil.login will be called. In new logic, only no exception, or exception is LoginException/InterruptedException, SecurityUtil.login() will be called.  If there is other exception, then behavior is different.

Is intended behavior change?

**Comment by yydzero** [comment]
Thanks a lot @asimrp to fix this inconsistent issue between gphdfs and pljava.

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java (hunk 1 above)
Comment (randomtask1155) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: i agree if there is some other exception thrown during login then kinitdisabled is skipped regardless of user config

Comment (yydzero) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: @randomtask1155  so do you mean we should keep the new code?

Comment (randomtask1155) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: @yydzero sorry i missed something before and somehow thought there would be a case where EX was not null and not a login exception or interrupted exception

The new code seems have similar behavior as before
looks good! 👍

Comment (Eulerizeit) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: Are we happy with going to Java 1.7 or do we want to go to 8. Ideally we
will only rebase the PL/Java stuff once.

--Rob

**Comment by Eulerizeit** [comment]
There is no need to refactor gphdfs unless it is beneficial to the performance of gphdfs. PL/Java supports versions 8, 7, and 6. We just need to agree on what we are using and then use it. If we want Java 7 or 8 just ask.

File: gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java (hunk 1 above)
Comment (yydzero) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: @randomtask1155 yes, when kinitDisabled is false, and ES is ABCException, then old code will call SecurityUtil.login, while new code will not call SecurityUtil.login.

Comment (yydzero) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: @Eulerizeit  if we agree to upgrade, then +1 for java8. Also GPText 2.0 will require Java 8.

Comment (davecramer) [inline gpAux/extensions/gphdfs/src/java/1.2/com/emc/greenplum/gpdb/hdfsconnector/ConnectorUtil.java]: pl/java has already been rebased. It is somewhat trivial to build it with 1.7 for sure. 1.8 I'm not sure of.

**Comment by asimrp** [comment]
Thank you all for the review.  I agree with @Eulerizeit, if pljava that can be built with JDK 1.7, the change in this PR is not necessary.  My confusion was due to a stale copy of pljava we keep in gpAux/extensions/pljava.  Removing / updating this copy allows us to consistently use JDK 1.7 and above to build gpdb.  Comments?

**Comment by yydzero** [comment]
+1 for updating pljava

**Comment by asimrp** [comment]
Closing this PR without merging in favour of refreshing `gpAux/extensions/pljava`.

**Pull request closed without merging**